meshtastic/tests/test_node_runtime_transport.py

```
@@ -0,0 +1,1294 @@
+                ch.role = channel_pb2.Channel.Role.SECONDARY
+            else:
+                ch.role = channel_pb2.Channel.Role.DISABLED
+            channels.append(ch)
+        mock_local_node.channels = channels
+
+        with patch.object(
+            delete_channel_runtime,
+            "_execute_rewrite_plan",
+            side_effect=RuntimeError("rewrite failed"),
+        ):
+            with pytest.raises(RuntimeError, match="rewrite failed"):
```

Ruff (0.15.6)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)